website/docs/developer/research/research-lambda-container.md

```diff
@@ -0,0 +1,139 @@
+- you can now package and deploy Lambda functions as container images of up to 10 GB in size compared to only 250MB in case of lambda deployment package size. 
+- preferable for data-heavy or dependency-heavy applications.
+- The Lambda service provides a variety of base image options with pre-installed runtimes that will be patched and maintained by AWS.
+- There is no additional cost to use the container feature but you have to pay for the ECR repository.
+
+### Cons
+- Container support requires your Lambda function code point to an ECR Repo URI which means that repo also has to be maintained. This also includes additional cost to use the ECR service.
+- need additional work to create the Dockerfile, tag and push the image to ECR repo.
+- Need additional work for deleting older images under the ECR repository.
+
+:::warning
+Currently NGAP does not support using ECR repository to store images which could bring possible risk and challenges to using containers for lambdas.
+:::
+
+## New configuration for lambda container
+There are a few configuration that needs to be added if the lambda is deployed from docker image.
+- Architecture- This  determines the type of computer processor that Lambda uses to run the function. Lambda provides a choice of instruction set architectures which is either `arm64` or `x86_64`. The `arm64` architecture offer lower cost per Gb/s compared to the other one. Check this [link](https://docs.aws.amazon.com/lambda/latest/dg/foundation-arch.html?icmpid=docs_lambda_help) for additional information.
+- Image configuration- These are values that can be used to override the container image settings for `ENTRYPOINT`, `CMD`, `WORKDIR` and `ENV`.
+
+```
+  ENTRYPOINT – Specifies the absolute path of the entry point to the application.
+
+  CMD – Specifies parameters that you want to pass in with ENTRYPOINT.
+
+  WORKDIR – Specifies the absolute path of the working directory.
+
+  ENV – Specifies an environment variable for the Lambda function.
+```
+- Image URI- The location of the container image to use for your function.
+
+
+## Creating a prototype
+Creating or updating the function is done by building a Docker image, uploading the new version to ECR and deploying/updating the Lambda function to point to the newly uploaded image using terraform. Docker CLI has been used to perform the task.
+
+The steps for prototyping lambda container are as follows:
+- Create or use an existing AWS ECR repository to store the Docker image. 
+- Create the Dockerfile and then build, tag and push the image to the above ECR repository.
+- Update terraform configuration of the lambda function to deploy it as container.
+
+:::warning
+Currently NGAP only allows developers to create a `private` ECR repository which was used here to create the prototype. Using a `public` ECR repository will need approval from NGAP first which could bring security concerns.
```

Comment:
Theoretically, could we create a private ECR repository for each installation?

Reply:
yes I think so as long as the repo names are different.